### ISSUE TITLE
A11y : Cannot expand CollapsibleSection via keyboard

### ISSUE THREAD

**Issue by zuchka** [grafana/grafana#46503]
**Steps to reproduce**:

try to access the collapsible sections in the new unified alerting UI: 

`http://localhost:3000/alerting/notifications/receivers/new?alertmanager=grafana`

![CleanShot 2022-03-11 at 16 55 26@2x](https://user-images.githubusercontent.com/37156449/158001091-0ffa5f23-17b5-4d5f-a3e9-6e2fa8a56c29.png)

**Actual Result**:

Various Unified Alerting elements are nested inside ["CollapsibleSection"](https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/components/receivers/form/CollapsibleSection.tsx#L18-L28) components. you cannot expand these dropdowns using the keyboard. Once expanded, the elements are focusable.

Cannot expand section via keyboard

**Expected Result**

workflow accessible for keyboard-only users

**Relevant WCAG Criteria:**

[2.1.1 Keyboard - Level A](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0&showtechniques=335%2C332#keyboard)

**Environment**:
- Grafana version: 8.4.3
- Data source type & version:
- User OS & Browser:
- Others:
